daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/CharsetUtils.scala

```diff
@@ -33,11 +33,10 @@ object CharsetUtils {
    * encodings as well as the standard ones.
    */
   def getCharset(name: String): BitsCharset = {
-    val cs = DaffodilCharsetProvider.charsetForName(name)
-    cs
+    CharsetCompilerRegistry.find(name).compileCharset.newInstance()
```

Comment:
Will become

```scala
    BitsCharsetDefinitionRegistry.find(name).newFactory().newInstance()
```

Not sure if Scala conventions would keep empty parentheses or not.